sys/include/net/ng_rpl/structs.h

```diff
@@ -0,0 +1,234 @@
+
+/**
+ * @ingroup net_ng_rpl
+ * @{
+ *
+ * @file        ng_rpl_structs.h
+ * @brief       RPL data structs
+ *
+ * Header file, which defines all structs used by RPL.
+ *
+ * @author      Eric Engel <eric.engel@fu-berlin.de>
+ * @author      Cenk Gündoğan <cnkgndgn@gmail.com>
+ */
+
+#ifndef NG_RPL_STRUCTS_H_
+#define NG_RPL_STRUCTS_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "net/ng_ipv6.h"
+#include "trickle.h"
+
+/**
+ * @brief RPL-Option Generic Format
+ * @see <a href="https://tools.ietf.org/html/rfc6550#section-6.7.1">
+ *          RPL Control Message Option Generic Format
+ *      </a>
+ */
+typedef struct __attribute__((packed)) {
+    uint8_t type;       /**< Option Type */
+    uint8_t length;     /**< Option Length, does not include the first two byte */
+} ng_rpl_opt_t;
+
+/**
+ * @brief DIO Base Object
+ * @see <a href="https://tools.ietf.org/html/rfc6550#section-6.3.1">
+ *          Format of the DIO Base Object
+ *      </a>
+ */
+typedef struct __attribute__((packed)) {
+    uint8_t instance_id;        /**< id of the isntance */
+    uint8_t version_number;     /**< version number of the DODAG */
+    network_uint16_t rank;      /**< rank of the parent emitting the DIO */
+    uint8_t g_mop_prf;          /**< grounded, MOP, preferred flags */
+    uint8_t dtsn;               /**< Destination Advertisement Trigger Sequence Number */
+    uint8_t flags;              /**< unused */
+    uint8_t reserved;           /**< reserved */
+    ng_ipv6_addr_t dodag_id;    /**< id of the dodag */
+} ng_rpl_dio_t;
+
+/**
+ * @brief DODAG Configuration Option
+ * @see <a href="https://tools.ietf.org/html/rfc6550#section-6.7.6">
+ *          DODAG Configuration
+ *      </a>
+ */
+typedef struct __attribute__((packed)) {
+    uint8_t type;                       /**< Option Type: 0x04 */
+    uint8_t length;                     /**< length of option, not including first two bytes */
+    uint8_t flags_a_pcs;                /**< flags */
+    uint8_t DIOIntDoubl;                /**< trickle Imax parameter */
+    uint8_t DIOIntMin;                  /**< trickle Imin parameter */
+    uint8_t DIORedun;                   /**< trickle k parameter */
+    network_uint16_t MaxRankIncrease;   /**< allowable increase in rank */
+    network_uint16_t MinHopRankIncrease;/**< DAGRank(rank) = floor(rank/MinHopRankIncrease) */
+    network_uint16_t ocp;               /**< Objective Code Point */
+    uint8_t reserved;                   /**< reserved */
+    uint8_t default_lifetime;           /**< lifetime of RPL routes (lifetime * lifetime_unit) */
+    network_uint16_t lifetime_unit;     /**< unit in seconds */
+} ng_rpl_opt_dodag_conf_t;
+
+/**
+ * @brief DODAG Information Solicitation
+ * @see <a href="https://tools.ietf.org/html/rfc6550#section-6.2">
+ *          DODAG Information Solicitation
+ *      </a>
+ */
+typedef struct __attribute__((packed)) {
+    uint8_t flags;      /**< unused */
+    uint8_t reserved;   /**< reserved */
```

Comment:
true .. unused and reserved do seem really important (:
BTW, I do not understand your indentation hints here and everywhere else, because for me the indent levels are correct